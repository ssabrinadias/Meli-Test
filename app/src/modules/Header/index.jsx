import React, { useState } from "react"
import styled from 'styled-components'
import { theme } from 'styled-tools'
import {Link} from "react-router-dom";  
import Search from "../Search"
import Content from "../../components/Content"
import LogoImage from "../../assets/images/Logo_ML_2x.png"


const HeaderContent = styled.header`
    display: flex;
    justify-content: center;
    padding: 5px 0;
    background: ${theme('colors.primary.main')};
`
const Logo = styled.a`
    display: flex; 
    align-items: center;
    width: 50px;
    height: 40px;
    margin-right: 20px;
    img {
        display: block;
        width: 100%;
        height: auto;
    }

`
const Box = styled.div`
    display: flex;
`

const Header = () =>(
    <HeaderContent>
        <Content >
            <Box>
                <Logo href="/">
                    <img src={LogoImage} alt="Logo Meli"/>
                </Logo>
                <Search/>
            </Box>
        </Content>
    </HeaderContent>
)

export default Header;