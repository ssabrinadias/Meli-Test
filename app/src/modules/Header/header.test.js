import React from "react";
import { render } from "@testing-library/react";
import Header from ".";

describe("Header component", () => {
	it("[snapshot] - should be able to render ", () => {
		const { container } = render(<Header />);
		expect(container).toMatchSnapshot();
	});
});
