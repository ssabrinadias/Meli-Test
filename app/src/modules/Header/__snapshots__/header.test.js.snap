// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component [snapshot] - should be able to render  1`] = `
<div>
  <header
    class="sc-AxgMl isISaG"
  >
    <div
      class="sc-AxhUy eqTxEv"
    >
      <div
        class="sc-Axmtr hIHYcU"
      >
        <a
          class="sc-AxheI gyZAte"
          href="/"
        >
          <img
            alt="Logo Meli"
            src="[object Object]"
          />
        </a>
        <form
          class="sc-AxirZ hvoVdz"
        >
          <input
            class="sc-AxiKw iculOt"
            name="search"
            placeholder="Nunca dejes de Buscar"
            required=""
            type="text"
            value=""
          />
          <button
            class="sc-AxhCb kBUfZc"
            type="submit"
          />
        </form>
      </div>
    </div>
  </header>
</div>
`;
